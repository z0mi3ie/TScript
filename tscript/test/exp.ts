//
// test precedence and associativity of operators
//

print 1+2*3;
print 3*2+1;
print (1+2)*3;
print 1+2+3;
print 2*3*4;

var abc;
var def;

print abc = def = 42;
print abc + def;

print abc = 1 + 2 * 3;

