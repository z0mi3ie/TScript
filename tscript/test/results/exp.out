7
7
9
6
24
42
84
7
